app/src/main/res/xml/settings.xml

```diff
@@ -13,9 +13,9 @@
             android:title="Game Settings"
         >
         <SwitchPreference
-            android:key="hide_ranks"
-            android:title="Hide player ranks"
-            android:defaultValue="false"
+            android:key="show_ranks"
+            android:title="Show player ranks"
+            android:defaultValue="true"
```

Comment:
You have the checkbox on by default, but the SettingsRepository off by default. It needs to be true in both places.

Reply:
true, forgot to invert this one when going from "hide" to "show", good catch